pandas/core/dtypes/missing.py

```diff
@@ -231,7 +231,11 @@ def _isna_ndarraylike(obj):
 
 
 def _isna_ndarraylike_old(obj):
-    values = getattr(obj, "_values", obj)
+    if not isinstance(obj, np.ndarray):
```

Comment:
is the the same construct we use in isna_arraylike?

Reply:
In _isna_arraylike it looks like we use getattr, but then check if we have an ExtensionArray and call the na method if so

Reply:
I agree the code should use the same structure as `_isna_ndarraylike`, refactoring out the common code maybe advantageous to avoid divergence occurring again.

Reply:
How would you recommend using a similar pattern here? We can't rely on isna to catch infinity (it only works if you construct the array _after_ setting the option):

```python
[ins] In [1]: cat = pd.Categorical([1, 2, np.inf])

[ins] In [2]: pd.options.mode.use_inf_as_na = True

[ins] In [3]: cat.isna()
Out[3]: array([False, False, False])

[ins] In [4]: cat = pd.Categorical([1, 2, np.inf])

[ins] In [5]: cat.isna()
Out[5]: array([False, False,  True])
```

Reply:
so is there also a bug in Categorical.isna?

Reply:
I suppose you could call this behavior a bug, but I'm not sure if it's a problem with Categorical.isna (it's simply checking Categorical._codes, which makes total sense and shouldn't need to change in my opinion). IMHO the issue is more with trying to have the behavior of these constructors / methods respect hidden global variables instead of being well-defined in isolation (I could even see the merit of deprecating options like this rather than needing to patch every place where NA checking might happen)

Reply:
I would prefer to make this code substatially the same, difference only be the inf change.